File: questPresentation/src/main/java/ru/nekit/android/qls/quest/answer/TrafficLightQuestAnswerVariantAdapter.java
package ru.nekit.android.qls.quest.answer;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import ru.nekit.android.qls.quest.answer.common.IQuestAnswerVariantAdapter;
import ru.nekit.android.qls.quest.model.TrafficLightModel;

public class TrafficLightQuestAnswerVariantAdapter implements IQuestAnswerVariantAdapter<Integer> {

    @Override
    @Nullable
    public String adapt(@NonNull Context context, @NonNull Integer answerVariant) {
        return TrafficLightModel.fromOrdinal(answerVariant).getString(context);
    }


}

File: questPresentation/src/main/java/ru/nekit/android/qls/quest/answer/TrafficLightQuestAnswerVariantAdapter.kt
package ru.nekit.android.qls.quest.answer

import android.content.Context
import ru.nekit.android.qls.data.representation.getRepresentation
import ru.nekit.android.qls.domain.model.resources.TrafficLightResourceCollection

class TrafficLightQuestAnswerVariantAdapter : IAnswerVariantAdapter<Int> {

    override fun adapt(context: Context, answerVariant: Int): String? {
        return TrafficLightResourceCollection.getById(answerVariant).getRepresentation().getRandomString(context)
    }

}
